### PR TITLE
Add raise_on_ratelimit option to surface Zendesk rate limit info to callers

### PR DESCRIPTION
**Pull request by japaz**
Closes #687

## Summary

- Add `raise_on_ratelimit=True` constructor parameter that raises a new `RateLimitError` exception immediately on HTTP 429 instead of silently sleeping and retrying
- The exception carries `retry_after` (seconds from the `Retry-After` header) and the raw `response` object, so callers can handle retries themselves
- Enrich the existing `RatelimitBudgetExceeded` exception with `retry_after` and `response` attributes for parity
- Fully backward compatible: default behavior (sleep and retry) is unchanged

## Motivation

The current rate limit handling blocks the calling thread with `sleep()`, which is problematic for users running Zenpy inside Celery, RQ, or other job queue systems. With `raise_on_ratelimit=True`, callers can catch `RateLimitError` and reschedule the work using the job queue's native retry mechanism:

```python
from zenpy.lib.exception import RateLimitError

@app.task(bind=True, max_retries=5)
def sync_tickets(self):
    client = Zenpy(raise_on_ratelimit=True, ...)
    try:
        tickets = client.tickets()
    except RateLimitError as e:
        raise self.retry(countdown=e.retry_after)
```

## Changes

| File | Change |
|---|---|
| `zenpy/lib/exception.py` | New `RateLimitError` exception; enriched `RatelimitBudgetExceeded` with `retry_after`/`response` kwargs |
| `zenpy/__init__.py` | New `raise_on_ratelimit` parameter threaded into config |
| `zenpy/lib/api.py` | `BaseApi` accepts and stores `raise_on_ratelimit` (with `False` default for backward compat); `_call_api()` raises on 429 when enabled; `_parse_retry_after()` for defensive header parsing; `check_ratelimit_budget()` passes remaining seconds to exception |
| `tests/test_api/test_ratelimit.py` | 17 unit tests covering raise behavior, default behavior, budget enrichment, Retry-After parsing edge cases, and exception attributes |

## Test plan

- [x] All 17 tests pass (`pytest tests/test_api/test_ratelimit.py`)
- [x] Default behavior (raise_on_ratelimit=False) is preserved and tested
- [x] RatelimitBudgetExceeded backward compatibility tested (no kwargs still works)
- [x] Defensive Retry-After parsing handles non-integer and missing values

### CODE REVIEW

File: zenpy/lib/api.py [hunk 1]
```diff
@@ -69,13 +69,15 @@ class BaseApi(object):
     """
 
     def __init__(self, subdomain, session, timeout, ratelimit,
-                 ratelimit_budget, ratelimit_request_interval, cache, domain):
+                 ratelimit_budget, ratelimit_request_interval,
+                 raise_on_ratelimit, cache, domain):
```
[the remark this answers is not in the document]
Comment (japaz) [reply zenpy/lib/api.py]: Fixed in 340456a. Added `raise_on_ratelimit=False` as the default in the `BaseApi.__init__` signature (also defaulted `cache` and `domain` per the suggestion).
